core/src/main/java/org/infinispan/protostream/descriptors/Descriptor.java

```diff
@@ -163,17 +163,20 @@ private void setContainingType(Descriptor containingType) {
    }
 
    @Override
-   protected AnnotationConfiguration getAnnotationConfig(String annotationName) {
-      AnnotationConfiguration annotationConfiguration = getAnnotationsConfig().annotations().get(annotationName);
+   protected AnnotationConfiguration getAnnotationConfig(AnnotationElement.Annotation annotation) {
+      AnnotationConfiguration annotationConfiguration = getAnnotationsConfig().annotations().get(annotation.getName());
       if (annotationConfiguration == null) {
          return null;
       }
+      if (annotation.getPackageName() != null && !annotation.getPackageName().equals(annotationConfiguration.packageName())) {
+         return null;
+      }
       for (AnnotationElement.AnnotationTarget t : annotationConfiguration.target()) {
          if (t == AnnotationElement.AnnotationTarget.MESSAGE) {
             return annotationConfiguration;
          }
       }
-      throw new DescriptorParserException("Annotation '" + annotationName + "' cannot be applied to message types.");
+      throw new DescriptorParserException("Annotation '" + annotation.getName() + "' cannot be applied to message types.");
```

Comment:
you are using `annotation` in the other descriptors. 
```suggestion
      throw new DescriptorParserException("Annotation '" + annotation + "' cannot be applied to message types.");
```

Reply:
Done